tpm_quote/src/verify.rs

```diff
@@ -0,0 +1,117 @@
+    traits::Marshall,
+};
+
+pub(crate) fn verify_quote_signature(
+    attestation_data: &Attest,
+    signature: &Signature,
+    pk: &Public,
+) -> Result<Vec<u8>, Error> {
+    if attestation_data.attestation_type() != AttestationType::Quote {
+        return Err(Error::VerificationError(
+            "attestation is not a quote".to_owned(),
+        ));
+    }
+
+    let message = attestation_data.marshall()?;
+    let signature = match signature {
+        Signature::EcDsa(ecc_sig) => Ok(ecc_sig),
+        _ => Err(Error::VerificationError(
+            "signature is not ECDSA".to_owned(),
+        )),
+    }?;
+    let nonce = attestation_data.extra_data().to_vec();
+
+    let (r, s): ([u8; 32], [u8; 32]) = (
+        signature
+            .signature_r()
+            .value()
+            .try_into()
+            .map_err(|_| Error::VerificationError("failed to get R in signature".to_owned()))?,
+        signature
+            .signature_s()
+            .value()
+            .try_into()
+            .map_err(|_| Error::VerificationError("failed to get S in signature".to_owned()))?,
+    );
+
+    let (x, y) = match pk {
+        Public::Ecc {
+            object_attributes: _,
+            name_hashing_algorithm,
+            auth_policy: _,
+            parameters,
+            unique,
+        } => {
+            if *name_hashing_algorithm != HashingAlgorithm::Sha256 {
+                return Err(Error::VerificationError(
+                    "only SHA256 is supported".to_owned(),
+                ));
+            }
+
+            if parameters.ecc_curve() != EccCurve::NistP256 {
+                return Err(Error::VerificationError(
+                    "only NIST P-256 curve is supported".to_owned(),
+                ));
+            }
+
+            match parameters.ecc_scheme() {
+                EccScheme::EcDsa(hash_scheme) => {
+                    if hash_scheme.hashing_algorithm() != HashingAlgorithm::Sha256 {
+                        return Err(Error::VerificationError(
+                            "unsupported hash scheme in ECDSA".to_owned(),
+                        ));
+                    }
+                }
+                _ => {
+                    return Err(Error::VerificationError(
+                        "unsupported signature scheme".to_owned(),
+                    ));
+                }
+            }
+
+            let (x, y): ([u8; 32], [u8; 32]) = (
+                unique.x().value().try_into().map_err(|_| {
+                    Error::VerificationError("failed to get X in public key".to_owned())
+                })?,
+                unique.y().value().try_into().map_err(|_| {
+                    Error::VerificationError("failed to get Y in public key".to_owned())
+                })?,
+            );
+
+            Ok((x, y))
+        }
+        _ => Err(Error::VerificationError(
+            "unsupported public key".to_owned(),
+        )),
+    }?;
+
+    let mut point = [0x00u8; 65];
```

Comment:
Create a const variable for `65` ?

Reply:
Removed in c00096d152de059ca8b52dcb2b2d3c185ca9ed03.